docuvision/batchsendfulldir.py

```diff
@@ -0,0 +1,204 @@
+    with open(file, 'rb') as f:
+        raw_bytes = f.read()
+        enc_bytes = base64.encodebytes(raw_bytes)
+    enc_string = enc_bytes.decode("utf-8")
+    md5_sum = hashlib.md5(raw_bytes).hexdigest()
+    headers = {"x-api-key": APITOKEN }
+    req = {
+        "name": "customername_job_x",
+        "request": {
+            "document": {
+                "name": filepath.name,
+                "dataType": "blob",
+                "encodingType": "base64/utf-8",
+                "mimeType": filepath.suffix[1:], #removes the leading period
+                "model": "base-medrec-anesthesia", #modify this to the model you want to consume
+                "confidenceInterval": 0.7, #modify this to your liking
+                "isPerformOCR": True, #if set your result will have OCR'd words and bounding boxes
+                "sizeBytes": len(enc_bytes),
+                "md5Sum": md5_sum,
+                "data": enc_string,
+            }
+        }
+    }
+    resp = requests.request(
+        method="POST",
+        url=f"{APIADDRESS}", #docuvision/v1/tasks/7",
+        headers=headers,
+        data=json.dumps(req),
+        timeout=timeout
+    )
+    rjson = json.loads(resp.content)
+    print(resp.content)
+    req_id = rjson.get("id")
+    req_meta = rjson.get("metadata", {})
+    if 'apiKey' in req_meta.keys(): req_meta['apiKey']='hidden'
+    logging.info(f"Job posted. status={resp.status_code}. id={req_id}. file={filepath.name}. respmeta={req_meta}")
+
+    if req_id is not None: 
+        #write the job id of the newly created job to local file 
+        with open('pendingjobs.docuvision', 'a') as f:
+            f.write(f"{req_id} {filepath}\n")
+    return req_id
+
+#gets the api response for a given jobid
+def hankai_get_results(jobid):
+    logging.debug("Getting {}{}".format(APIADDRESS, jobid))
+    headers = {"x-api-key": APITOKEN }
+    resp = requests.request(
+        method="GET",
+        url=f"{APIADDRESS}{jobid}",
+        headers=headers
+    )
+    return resp
+
+#returns 1 if the job is complete (even if state is error) or 0 if not found or in progress
+def hankai_check_job_complete(resp):
+    try:
+        rjson= json.loads(resp.content)
+        if (rjson.get("state") and rjson.get("state") != "In-Progress"):
+            return 1
+    except: 
+        logging.error(f"in hank_check_job_complete(resp). resp={resp}")
+    return 0
+
+#writes out the results to a .json file of same base filestem+_jobid as original file
+def hankai_write_json_results(jobid, document_filepath, apiresponse):
+    dfP = Path(document_filepath)
+    jsonfp = dfP.parent / (dfP.stem + f'_{jobid}.json')
+    logging.debug(f"Writing api response for jobid={jobid} to {jsonfp}")
+    with open(jsonfp, 'w') as f:
+        jsonapir = json.loads(apiresponse.content)
+        jsonapir['metadata']['apiKey']="{}...".format(jsonapir['metadata']['apiKey'][:15])
+        json.dump(jsonapir, f, indent=2)
+
+#%% SUBMIT JOBS
+if 1:
```

Comment:
Can I only check on the jobs without submitting?

Reply:
that's the intent there. i'll make it a cmd parameter instead of an if 1 :)